src/net/rpc.rs

```diff
@@ -2384,6 +2390,110 @@ impl ConversationHttp {
         response.send(http, fd).and_then(|_| Ok(stream))
     }
 
+    /// Handle a GET health.
+    /// A node is considered healthy if it is up to the max height of its bootstrap nodes.
+    /// This endpoint also returns what percent of the max height the node height is at.
+    fn handle_get_health<W: Write>(
+        http: &mut StacksHttp,
+        fd: &mut W,
+        req: &HttpRequestType,
+        network: &PeerNetwork,
+        network_id: u32,
+        peerdb: &PeerDB,
+        block_stats: Option<&HashMap<NeighborKey, NeighborBlockStats>>,
+        canonical_stacks_tip_height: u64,
+    ) -> Result<(), net_error> {
+        let response_metadata =
+            HttpResponseMetadata::from_http_request_type(req, Some(canonical_stacks_tip_height));
+
+        let peer_version = network.peer_version;
+        let stacks_tip_height = network.burnchain_tip.canonical_stacks_tip_height;
+        let last_processed_burn_height = network.burnchain_tip.block_height;
+        let burnchain_height = network.chain_view.burn_block_height;
+        let epoch = network.get_current_epoch();
+
+        let initial_neighbors = PeerDB::get_valid_initial_neighbors(
+            peerdb.conn(),
+            network_id,
+            epoch.network_epoch,
+            peer_version,
+            burnchain_height,
+        )
+        .unwrap();
+
+        // no bootstrap nodes found, unable to determine health.
+        if initial_neighbors.len() == 0 {
+            let response = HttpResponseType::GetHealthNoDataError(
+                response_metadata.clone(),
+                "No viable bootstrap peers found, unable to determine health".to_string(),
+            );
+            return response.send(http, fd).map(|_| ());
+        }
+
+        let ibd = infer_initial_burnchain_block_download(
+            &network.burnchain,
+            last_processed_burn_height,
+            burnchain_height,
+        );
+
+        // get max block height amongst bootstrap nodes
+        let response = if let Some(block_stats) = block_stats {
```

Comment:
I think this code block should be part of the `InvState` state machine.  The code here should just call it in order to get the maximum block height reported by a given set of nodes.